pallets/operations/src/impls.rs

```diff
@@ -1,45 +1,56 @@
 #![allow(clippy::nonminimal_bool)]
 
-use frame_support::{dispatch::DispatchResult, traits::LockIdentifier, WeakBoundedVec};
-use pallet_balances::BalanceLock;
+use frame_support::dispatch::DispatchResult;
 use parity_scale_codec::Encode;
 use sp_core::hexdisplay::HexDisplay;
 use sp_runtime::DispatchError;
 
 use crate::{
     pallet::{Config, Event, Pallet},
-    traits::{AccountInfoProvider, BalancesProvider, BondedStashProvider, NextKeysSessionProvider},
-    LOG_TARGET, STAKING_ID,
+    traits::{
+        AccountInfoProvider, BalancesProvider, BondedStashProvider, ContractInfoProvider,
+        NextKeysSessionProvider,
+    },
+    LOG_TARGET,
 };
 
 impl<T: Config> Pallet<T> {
-    /// Checks if account has an underflow of `consumers` counter. In such case, it increments
-    /// it by one.
-    pub fn fix_underflow_consumer_counter(who: T::AccountId) -> DispatchResult {
+    /// Calculate expected consumers counter for a `who` account, and if actual
+    /// counter is not as expected, increment or decrement current counter
+    pub fn fix_consumer_counter(who: T::AccountId) -> DispatchResult {
         let current_consumers = T::AccountInfoProvider::get_consumers(&who);
         let mut expected_consumers: u32 = 0;
 
         if Self::reserved_or_frozen_non_zero(&who) {
             expected_consumers += 1;
         }
-        let has_staking_lock = Self::has_staking_lock(&who);
-        if has_staking_lock {
+        if Self::is_contract_account(&who) {
+            expected_consumers += 1;
+        }
+        if Self::is_bonded(&who) {
             expected_consumers += 1;
         }
         if Self::has_next_session_keys_and_account_is_controller(&who) {
             expected_consumers += 1;
         }
 
+        #[allow(clippy::comparison_chain)]
```

Comment:
Why? with match curretnt_consumers.cmp(expected_consumers) it would look better :P

Reply:
I tried really to get this match done, but in the end I did not like it like if you look at `x.cmp(&y)`, without checking docs, which order is it, `x > y`, or `< y`? Can reader tell in 1 ms looking at the line of code? They definitely do when they look in the current version. :) 
```
    /// fn f(x: u8, y: u8) {
    ///      match x.cmp(&y) {
    ///          Ordering::Greater => a(),
    ///          Ordering::Less => b(),
    ///          Ordering::Equal => c()
    ///      }
    /// }
```